javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/TypeExtractor.java

```diff
@@ -504,4 +505,44 @@ public ResolvedType visit(FieldDeclaration node, Boolean solveLambdas) {
         }
         throw new IllegalArgumentException("Cannot resolve the type of a field with multiple variable declarations. Pick one");
     }
+
+    @Override
+    public ResolvedType visit(SwitchExpr switchExpr, Boolean solveLambdas) {
+        List<ResolvedType> allSwitchEntiesResolved = switchExpr.getEntries().stream()
+                .map(entry -> entry.accept(this, solveLambdas))
+                .collect(toList());
+        // TODO find common type of all switch entries
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public ResolvedType visit(SwitchEntry entry, Boolean solveLambdas) {
+        switch (entry.getType()) {
+            case STATEMENT_GROUP:
+            case BLOCK:
+                return findCommonTypeOfAllBreakStatements(entry, solveLambdas);
+            case EXPRESSION:
+                return entry.getStatement(0).asExpressionStmt().getExpression().accept(this, solveLambdas);
+            case THROWS_STATEMENT:
+                // TODO is this correct?
+                return ResolvedVoidType.INSTANCE;
+            default:
+                throw new IllegalStateException("This kind of switch entry is not supported yet.");
+        }
+    }
+
+    private ResolvedType findCommonTypeOfAllBreakStatements(Node node, Boolean solveLambdas) {
+        List<ResolvedType> allBreakStmtsResolved = node.findAll(BreakStmt.class).stream()
+                .map(bs -> bs.accept(this, solveLambdas))
+                .collect(toList());
+        // TODO find common type of all break statements
```

Comment:
Again, we should have an utility for that somewhere. And if not we should create it

Reply:
I'll go hunt for it then.